Add note for air-gapped environments for pulling the Support diag image (#6021)
elastic/cloud-on-k8s: docs/operating-eck/troubleshooting/take-eck-dump.asciidoc

By default, the tool automatically runs link:https://github.com/elastic/support-diagnostics[support diagnostics] for every Elasticsearch cluster and Kibana instance. This requires the temporary deployment of additional Pods into the Kubernetes cluster. If this is not what you want, you can turn off the feature by specifying the `--run-stack-diagnostics=false` flag.

> Note that the ECK diagnostics will spawn a pod `docker.elastic.co/eck-dev/support-diagnostics:x.y.z` to gather Elasticsearch cluster diagnostics, but in air-gapped environments with no access to the `docker.elastic.co` registry, you should copy the (latest) image to your internal image registry and then run the ECK-diagnostics with the additional flag `--diagnostic-image internal-image-for-support-diagnostic>` to get Elasticsearch cluster information.

Reviewer: ```suggestion
In air-gapped environments with no access to the `docker.elastic.co` registry, you should copy the latest support-diagnostic Docker image to your internal image registry and then run the tool with the additional flag `--diagnostic-image <custom-support-diagnostics-image-name>`.
```

Reviewer: Today there is no official Elastic support-diagnostics Docker image so we build and push one specifically for eck-diagnostics. The downside I see is that this image is not easily discoverable by users. I'm not sure how users will get the name of the latest image.

Author: Thanks @thbkrkr, duly noted. I think the notes as it stands should be fine now. For the image not easily being discoverable, maybe I can raise some issue somewhere else?
